Patch-Source: https://github.com/pallets/werkzeug/commit/341f6e208d5d3425d39b480ebdf6a555f852ff50
From 341f6e208d5d3425d39b480ebdf6a555f852ff50 Mon Sep 17 00:00:00 2001
From: Derek Higgins <derekh@redhat.com>
Date: Mon, 20 Feb 2023 15:19:59 +0000
Subject: [PATCH] test_exclude_patterns does not rely on sys.prefix

sys.prefix may be empty if nothing was imported from it
---
 tests/test_serving.py | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/tests/test_serving.py b/tests/test_serving.py
index b6dc2543..053ebe0e 100644
--- a/tests/test_serving.py
+++ b/tests/test_serving.py
@@ -4,7 +4,6 @@ import os
 import shutil
 import socket
 import ssl
-import sys
 from io import BytesIO
 from pathlib import Path
 
@@ -128,12 +127,15 @@ def test_windows_get_args_for_reloading(monkeypatch, tmp_path):
 @pytest.mark.filterwarnings("ignore::pytest.PytestUnraisableExceptionWarning")
 @pytest.mark.parametrize("find", [_find_stat_paths, _find_watchdog_paths])
 def test_exclude_patterns(find):
-    # Imported paths under sys.prefix will be included by default.
+    # Select a path to exclude from the unfiltered list, assert that it is present and
+    # then gets excluded.
     paths = find(set(), set())
-    assert any(p.startswith(sys.prefix) for p in paths)
+    path_to_exclude = next(iter(paths))
+    assert any(p.startswith(path_to_exclude) for p in paths)
+
     # Those paths should be excluded due to the pattern.
-    paths = find(set(), {f"{sys.prefix}*"})
-    assert not any(p.startswith(sys.prefix) for p in paths)
+    paths = find(set(), {f"{path_to_exclude}*"})
+    assert not any(p.startswith(path_to_exclude) for p in paths)
 
 
 @pytest.mark.filterwarnings("ignore::pytest.PytestUnraisableExceptionWarning")
-- 
2.46.2

